subnet/watch.go

```diff
@@ -78,7 +79,11 @@ func (lw *leaseWatcher) reset(leases []Lease) []Event {
 		for i, ol := range lw.leases {
 			if ol.Subnet.Equal(nl.Subnet) {
 				lw.leases = deleteLease(lw.leases, i)
-				found = true
+
+				// If the backend data has changed, send the added event to the backend
+				if bytes.Compare(ol.Attrs.BackendData, nl.Attrs.BackendData) == 0 {
```

Comment:
Shouldn't we move the deleteLease inside the condition as well?

Reply:
Sorry, been like a year since I looked at this. 

I don't think so, IIRC the logic for delete lease is arguably correct, where if the lease is exactly the same, it should be deleted from lw.leases so that it doesn't trigger the EventRemoved in https://github.com/coreos/flannel/blob/14d007ba03ece66e7b18d0a27963ae662598e709/subnet/watch.go#L96-L102

However, what this does, is if the Lease still exists, but the backend data has changed, it will trigger a new EventAdded so that the backend will receive an event with new BackendData and provision updates, such as a changed mac address for vxlan.